core_impl/src/main/java/io/opencensus/stats/MeasureToViewMap.java

```diff
@@ -90,18 +93,30 @@ synchronized void registerView(ViewDescriptor viewDescriptor, Clock clock) {
   }
 
   // Records stats with a set of tags.
-  synchronized void record(StatsContextImpl tags, MeasurementMap stats) {
-    for (MeasurementValue mv : stats) {
+  synchronized void record(StatsContextImpl tags, MeasureMap stats) {
+    Iterator<Measurement> iterator = stats.iterator();
+    while (iterator.hasNext()) {
+      Measurement measurement = iterator.next();
       Collection<MutableView> views =
-          mutableMap.get(mv.getMeasurement().getName());
+          mutableMap.get(measurement.getMeasure().getName());
+      Object value = measurement.match(new GetDoubleValueFunc(), new GetLongValueFunc());
```

Comment:
Consider making the a Double and having GetLongValueFunc() throw and exception.

Also, you don't really need a new instance of GetDoubleValueFunc() and GetLongValueFunc() for each use.